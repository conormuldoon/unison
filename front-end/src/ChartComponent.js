import { has } from 'lodash';
import PropTypes from 'prop-types';
import React, { useEffect, useState } from 'react';
import { VictoryAxis, VictoryChart, VictoryLine, VictoryTheme, VictoryZoomContainer } from 'victory';
import './App.css';
import { varMapping } from './Util';



const PERCENT = "Percentage";
const CELSIUS = "Celsius";

const PREVAL = "precipitation.value";

/**
 * A component for displaying line graphs.
 * 
 * @component
 * 
 */
function ChartComponent(props) {

  const [yVal, setYVal] = useState(undefined);
  const [yLabel, setYLabel] = useState(undefined);


  useEffect(() => {



    // 'Precipitation','Humidity','Wind Direction','Wind Speed','Cloudiness','Cloud Level','Dew Point','Pressure','Temperature'
    const variRequest = varMapping(props.varCur);

    if (variRequest === "precipitation") {
      if (props.minMax) {
        if (props.index === 0) {
          setYVal("precipitation.minvalue");
        } else if (props.index === 1) {
          setYVal(PREVAL);
        } else {
          setYVal("precipitation.maxvalue");
        }

      } else {
        setYVal(PREVAL);
      }
      setYLabel("Millimetres");

    } else if (variRequest === "humidity") {
      setYVal(variRequest);
      setYLabel(PERCENT);
    } else if (variRequest === "windDirection") {

      // Displaying sin of the angle on the graph, rather than the angle/
      setYVal("windDirection.sinDeg");
      setYLabel("sin of angle");
    } else if (variRequest === "windSpeed") {
      setYVal("windSpeed.mps");
      setYLabel("Metres per second");
    } else if (variRequest === "cloudiness") {
      setYVal(variRequest);
      setYLabel(PERCENT);
    } else if (variRequest === "cloudLevel") {
      if (props.index === 0) {
        setYVal("cloud.low");
      } else if (props.index === 1) {
        setYVal("cloud.medium");
      } else {
        setYVal("cloud.high");
      }
      setYLabel(PERCENT);
    } else if (variRequest === "dewPoint") {
      setYVal(variRequest);
      setYLabel(CELSIUS);
    } else if (variRequest === "pressure") {
      setYVal(variRequest);
      setYLabel("Hectopascals");
    } else if (variRequest === "temperature") {
      setYVal(variRequest);
      setYLabel(CELSIUS);
    }


  }, [props.varCur, props.minMax, props.index]);




  if (yVal && props.data && has(props.data[0], yVal)) {

    return (

      <VictoryChart style={{ parent: { maxWidth: "55%", maxHeight: "55%" } }} responsive={true} padding={{ left: 70, bottom: 50, top: 10 }}
        domainPadding={{ x: [0, 0], y: 0 }}
        theme={VictoryTheme.material}
        scale={{ x: "time" }}
        containerComponent={
          <VictoryZoomContainer
            zoomDimension="x"
            zoomDomain={props.zoomDomain}
            onZoomDomainChange={props.handleZoom}
          />
        }
      >
        <VictoryLine
          style={{
            data: { stroke: "#c43a31" },
            parent: { border: "1px solid #ccc" }
          }}
          data={props.data}
          x="date"
          y={yVal}
        />

        <VictoryAxis fixLabelOverlap={true} />
        <VictoryAxis dependentAxis label={yLabel} style={{ axisLabel: { padding: 45 } }} tickFormat={
          (x) => { if (x < .001) { return x.toFixed(3); } if (x > 999) { return Math.round(x); } return x; }} />

      </VictoryChart>

    );
  } else {

    return false;
  }

}


ChartComponent.propTypes = {
  /** Specifies the beginning and end times of the time domain of the x-axis. */
  zoomDomain: PropTypes.object,

  /** Called when there is a zoom domain change. */
  handleZoom: PropTypes.func.isRequired,

  /** Data to be displayed on the graph. */
  data: PropTypes.array,


  /** The weather variable currently selected. */
  varCur: PropTypes.string.isRequired,

  /** True if the if displaying tabs for the precipitation variable.  */
  minMax: PropTypes.bool,

  /** The tab index - 0 for the left tab, 1 for the middle tab, and 2 for the right tab. */
  index: PropTypes.number,

}

export default ChartComponent;


