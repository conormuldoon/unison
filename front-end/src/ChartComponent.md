ChartComponent example:

```js

const data=[{"date":"2019-02-23T19:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-23T20:00:00.000+0000","precipitation":{"value":0.5,"minvalue":0.0,"maxvalue":0.7}},{"date":"2019-02-23T21:00:00.000+0000","precipitation":{"value":0.6,"minvalue":0.0,"maxvalue":0.6}},{"date":"2019-02-23T22:00:00.000+0000","precipitation":{"value":3.0,"minvalue":0.0,"maxvalue":0.4}},{"date":"2019-02-23T23:00:00.000+0000","precipitation":{"value":0.2,"minvalue":0.0,"maxvalue":0.5}},{"date":"2019-02-26T18:00:00.000+0000","precipitation":{"value":0.7,"minvalue":0.0,"maxvalue":0.8}},{"date":"2019-02-25T12:00:00.000+0000","precipitation":{"value":0.3,"minvalue":0.0,"maxvalue":0.5}}];


<ChartComponent data={data} yVal={'precipitation.value'} yLabel={'Millimetres'} />


```  
