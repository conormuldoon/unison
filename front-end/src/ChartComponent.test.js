import React from 'react';
import ReactDOM from 'react-dom';
import ChartComponent from './ChartComponent';

import { render} from "react-testing-library"



const data=[{"date":"2019-02-23T19:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-23T20:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-23T21:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-23T22:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-23T23:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-26T18:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-25T12:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-25T13:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-25T14:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-25T15:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-25T16:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-25T17:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-25T18:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-25T19:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-25T20:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-25T21:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-25T22:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-25T23:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-26T00:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-26T01:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-26T02:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-26T03:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-26T04:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-26T05:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-26T06:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-26T07:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-26T08:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-26T09:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-26T10:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-26T11:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-26T12:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-26T13:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-26T14:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-26T15:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-26T16:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-26T17:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-24T00:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-24T01:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-24T02:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-24T03:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-24T04:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-24T05:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-24T06:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-24T07:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-24T08:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.1}},{"date":"2019-02-24T09:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.1}},{"date":"2019-02-24T10:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-24T11:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-24T12:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-24T13:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-24T14:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-24T15:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-24T16:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-24T17:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-24T18:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-24T19:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-24T20:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-24T21:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-24T22:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-24T23:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-25T00:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-25T01:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-25T02:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-25T03:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-25T04:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-25T05:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-25T06:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-25T07:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-25T08:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-25T09:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-25T10:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-25T11:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-27T00:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-27T01:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-27T02:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-27T03:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-27T04:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-27T05:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-27T06:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-27T07:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-27T08:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-27T09:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-27T10:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-27T11:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-27T12:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-27T13:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-27T14:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-27T15:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-27T16:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-27T17:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-27T18:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-27T19:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-27T20:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-27T21:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-27T22:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-27T23:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-28T00:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-28T01:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-28T02:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-28T03:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-28T04:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-28T05:00:00.000+0000","precipitation":{"value":0.2,"minvalue":0.1,"maxvalue":0.3}},{"date":"2019-02-26T19:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-26T20:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-26T21:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-26T22:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-26T23:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}}];

const chartComponent= <ChartComponent varCur='Precipitation' data={data} yVal={'precipitation.value'} yLabel={'Millimetres'} />;

it('renders without crashing', async () => {

  const div = document.createElement('div');

  ReactDOM.render(chartComponent,div);
  ReactDOM.unmountComponentAtNode(div);
});

it('mathes snapshot', () => {


  const {container} = render(chartComponent);

  expect(container).toMatchSnapshot();

});
