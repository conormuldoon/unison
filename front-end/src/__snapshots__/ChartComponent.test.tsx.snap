// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mathes snapshot 1`] = `
<div>
  <div
    class="VictoryContainer"
    style="max-width: 55%; max-height: 55%; height: 100%; width: 100%; user-select: none; pointer-events: none; position: relative;"
  >
    <svg
      height="350"
      role="img"
      style="pointer-events: all; width: 100%; height: auto;"
      viewBox="0 0 350 350"
      width="350"
    >
      <g
        clip-path="url(#victory-clip-4)"
        style="border: 1px solid #ccc; width: 100%; height: 100%;"
      >
        <defs>
          <clippath
            id="victory-clip-4"
          >
            <rect
              height="290"
              vector-effect="non-scaling-stroke"
              width="280"
              x="70"
              y="10"
            />
          </clippath>
        </defs>
        <path
          d="MNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,203.33333333333334LNaN,203.33333333333334LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300LNaN,10LNaN,300LNaN,300LNaN,300LNaN,300LNaN,300"
          role="presentation"
          shape-rendering="auto"
          style="fill: transparent; stroke: #c43a31; opacity: 1; stroke-width: 2;"
        />
      </g>
      <g
        role="presentation"
      >
        <line
          role="presentation"
          shape-rendering="auto"
          style="stroke: black;"
          vector-effect="non-scaling-stroke"
        />
      </g>
      <g
        role="presentation"
      >
        <line
          role="presentation"
          shape-rendering="auto"
          style="stroke: #90A4AE; fill: transparent; stroke-width: 2; stroke-linecap: round; stroke-linejoin: round;"
          vector-effect="non-scaling-stroke"
          x1="70"
          x2="70"
          y1="10"
          y2="300"
        />
        <text
          direction="inherit"
          dx="0"
          dy="-1.7400000000000002"
          id="chart-axis-2-axisLabel-0"
          transform="rotate(-90,25,155)"
          x="25"
          y="155"
        >
          <tspan
            dx="0"
            style="padding: 45px; text-anchor: middle; font-family: 'Roboto', 'Helvetica Neue', Helvetica, sans-serif; font-size: 12px; letter-spacing: normal; fill: #455A64; stroke: transparent; stroke-width: 0;"
            text-anchor="middle"
            x="25"
          >
            Millimetres
          </tspan>
        </text>
        <g
          role="presentation"
        >
          <line
            role="presentation"
            shape-rendering="auto"
            style="stroke: #ECEFF1; fill: none; stroke-dasharray: 10, 5; stroke-linecap: round; stroke-linejoin: round; pointer-events: painted;"
            vector-effect="non-scaling-stroke"
            x1="70"
            x2="350"
            y1="251.66666666666663"
            y2="251.66666666666663"
          />
          <line
            role="presentation"
            shape-rendering="auto"
            style="stroke: #90A4AE; fill: transparent; size: 5px; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
            vector-effect="non-scaling-stroke"
            x1="70"
            x2="65"
            y1="251.66666666666663"
            y2="251.66666666666663"
          />
          <text
            direction="inherit"
            dx="0"
            dy="4.26"
            id="chart-axis-2-tickLabels-0"
            x="57"
            y="251.66666666666663"
          >
            <tspan
              dx="0"
              style="font-family: 'Roboto', 'Helvetica Neue', Helvetica, sans-serif; font-size: 12px; letter-spacing: normal; padding: 8px; fill: #455A64; stroke: transparent; stroke-width: 0;"
              text-anchor="end"
              x="57"
            >
              0.05
            </tspan>
          </text>
        </g>
        <g
          role="presentation"
        >
          <line
            role="presentation"
            shape-rendering="auto"
            style="stroke: #ECEFF1; fill: none; stroke-dasharray: 10, 5; stroke-linecap: round; stroke-linejoin: round; pointer-events: painted;"
            vector-effect="non-scaling-stroke"
            x1="70"
            x2="350"
            y1="203.33333333333334"
            y2="203.33333333333334"
          />
          <line
            role="presentation"
            shape-rendering="auto"
            style="stroke: #90A4AE; fill: transparent; size: 5px; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
            vector-effect="non-scaling-stroke"
            x1="70"
            x2="65"
            y1="203.33333333333334"
            y2="203.33333333333334"
          />
          <text
            direction="inherit"
            dx="0"
            dy="4.26"
            id="chart-axis-2-tickLabels-1"
            x="57"
            y="203.33333333333334"
          >
            <tspan
              dx="0"
              style="font-family: 'Roboto', 'Helvetica Neue', Helvetica, sans-serif; font-size: 12px; letter-spacing: normal; padding: 8px; fill: #455A64; stroke: transparent; stroke-width: 0;"
              text-anchor="end"
              x="57"
            >
              0.1
            </tspan>
          </text>
        </g>
        <g
          role="presentation"
        >
          <line
            role="presentation"
            shape-rendering="auto"
            style="stroke: #ECEFF1; fill: none; stroke-dasharray: 10, 5; stroke-linecap: round; stroke-linejoin: round; pointer-events: painted;"
            vector-effect="non-scaling-stroke"
            x1="70"
            x2="350"
            y1="155"
            y2="155"
          />
          <line
            role="presentation"
            shape-rendering="auto"
            style="stroke: #90A4AE; fill: transparent; size: 5px; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
            vector-effect="non-scaling-stroke"
            x1="70"
            x2="65"
            y1="155"
            y2="155"
          />
          <text
            direction="inherit"
            dx="0"
            dy="4.26"
            id="chart-axis-2-tickLabels-2"
            x="57"
            y="155"
          >
            <tspan
              dx="0"
              style="font-family: 'Roboto', 'Helvetica Neue', Helvetica, sans-serif; font-size: 12px; letter-spacing: normal; padding: 8px; fill: #455A64; stroke: transparent; stroke-width: 0;"
              text-anchor="end"
              x="57"
            >
              0.15
            </tspan>
          </text>
        </g>
        <g
          role="presentation"
        >
          <line
            role="presentation"
            shape-rendering="auto"
            style="stroke: #ECEFF1; fill: none; stroke-dasharray: 10, 5; stroke-linecap: round; stroke-linejoin: round; pointer-events: painted;"
            vector-effect="non-scaling-stroke"
            x1="70"
            x2="350"
            y1="106.66666666666664"
            y2="106.66666666666664"
          />
          <line
            role="presentation"
            shape-rendering="auto"
            style="stroke: #90A4AE; fill: transparent; size: 5px; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
            vector-effect="non-scaling-stroke"
            x1="70"
            x2="65"
            y1="106.66666666666664"
            y2="106.66666666666664"
          />
          <text
            direction="inherit"
            dx="0"
            dy="4.26"
            id="chart-axis-2-tickLabels-3"
            x="57"
            y="106.66666666666664"
          >
            <tspan
              dx="0"
              style="font-family: 'Roboto', 'Helvetica Neue', Helvetica, sans-serif; font-size: 12px; letter-spacing: normal; padding: 8px; fill: #455A64; stroke: transparent; stroke-width: 0;"
              text-anchor="end"
              x="57"
            >
              0.2
            </tspan>
          </text>
        </g>
        <g
          role="presentation"
        >
          <line
            role="presentation"
            shape-rendering="auto"
            style="stroke: #ECEFF1; fill: none; stroke-dasharray: 10, 5; stroke-linecap: round; stroke-linejoin: round; pointer-events: painted;"
            vector-effect="non-scaling-stroke"
            x1="70"
            x2="350"
            y1="58.33333333333332"
            y2="58.33333333333332"
          />
          <line
            role="presentation"
            shape-rendering="auto"
            style="stroke: #90A4AE; fill: transparent; size: 5px; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
            vector-effect="non-scaling-stroke"
            x1="70"
            x2="65"
            y1="58.33333333333332"
            y2="58.33333333333332"
          />
          <text
            direction="inherit"
            dx="0"
            dy="4.26"
            id="chart-axis-2-tickLabels-4"
            x="57"
            y="58.33333333333332"
          >
            <tspan
              dx="0"
              style="font-family: 'Roboto', 'Helvetica Neue', Helvetica, sans-serif; font-size: 12px; letter-spacing: normal; padding: 8px; fill: #455A64; stroke: transparent; stroke-width: 0;"
              text-anchor="end"
              x="57"
            >
              0.25
            </tspan>
          </text>
        </g>
        <g
          role="presentation"
        >
          <line
            role="presentation"
            shape-rendering="auto"
            style="stroke: #ECEFF1; fill: none; stroke-dasharray: 10, 5; stroke-linecap: round; stroke-linejoin: round; pointer-events: painted;"
            vector-effect="non-scaling-stroke"
            x1="70"
            x2="350"
            y1="10"
            y2="10"
          />
          <line
            role="presentation"
            shape-rendering="auto"
            style="stroke: #90A4AE; fill: transparent; size: 5px; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
            vector-effect="non-scaling-stroke"
            x1="70"
            x2="65"
            y1="10"
            y2="10"
          />
          <text
            direction="inherit"
            dx="0"
            dy="4.26"
            id="chart-axis-2-tickLabels-5"
            x="57"
            y="10"
          >
            <tspan
              dx="0"
              style="font-family: 'Roboto', 'Helvetica Neue', Helvetica, sans-serif; font-size: 12px; letter-spacing: normal; padding: 8px; fill: #455A64; stroke: transparent; stroke-width: 0;"
              text-anchor="end"
              x="57"
            >
              0.3
            </tspan>
          </text>
        </g>
      </g>
    </svg>
    <div
      style="z-index: 99; position: absolute; top: 0px; left: 0px; width: 100%; height: auto;"
    >
      <svg
        height="350"
        style="overflow: visible; width: 100%; height: auto;"
        viewBox="0 0 350 350"
        width="350"
      />
    </div>
  </div>
</div>
`;
